### ISSUE TITLE
EmitC: Enable Async mode in ttnn backend

### ISSUE THREAD

**Issue by jnie-TT** [tenstorrent/tt-mlir#733]
Related: #697 #732

**Comment by skhatriTT** [comment]
@svuckovicTT @jnie-TT Any update on this one?

**Comment by svuckovicTT** [comment]
@jnie-TT do you know what this async mode is? I can't find any details in metal code :(

**Comment by jnie-TT** [comment]
Probably device async mode. Since emitc currently piggy backs on top of the device created in ttrt, this will just follow if async mode is enabled in ttrt, so should be ok to close

**Comment by svuckovicTT** [comment]
@jnie-TT the ttnn-standalone tool opens its own device, so it should support async as well - however I see async was removed with these new mesh device changes, so this is a non-issue now...

For my own learning though, can you ELI5 what the async mode is, on a high level?

**Comment by jnie-TT** [comment]
@svuckovicTT yeah async should be implicitly handled now so nothing extra needs to be done, we've actually removed all of this explicit async handling in runtime as well. 

Async mode used to be a device parameter that determines if operations should run asynchronously on that device. If you run an operation on a device with async mode = true, the operation will return immediately with a not-yet-populated tensor that will get populated eventually in the future. Essentially allows you to do stuff on host while the op is running in the meantime.
